doc/modules/clustering.rst

```diff
@@ -767,6 +767,20 @@ by black points below.
     The possibility to use custom metrics is retained;
     for details, see :class:`NearestNeighbors`.
 
+    This implementation is by default not memory efficient, constructing a full
```

Comment:
I'd reword this as `This implementation is by default not memory efficient because it constructs a full ...`